### PR TITLE
refactor(cli): use PathBuf instead of String for lint and fmt subcommands

### PR DESCRIPTION
**Pull request by humancalico**
There was not really a need for this PR but it was a pain to work with files with `String`s as you have to convert them every time you need to use them. So I went ahead and replaced the `String`s with `PathBuf`s (for the Lint and Fmt subcommands)

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -1872,7 +1878,10 @@ mod tests {
           files: vec![],
           rules: false,
           json: false,
-          ignore: svec!["script_1.ts", "script_2.ts"],
+          ignore: vec![
+            PathBuf::from("script_1.ts"),
+            PathBuf::from("script_2.ts")
+          ], // TODO(humancalico) maybe create a PathBuf macro like svec
```
Comment (bartlomieju) [inline cli/flags.rs:1884]: remove the TODO, I don't think a macro is needed

File: cli/fmt.rs [hunk 2]
```diff
@@ -32,11 +32,11 @@ const BOM_CHAR: char = '\u{FEFF}';
...
 /// then the current directory is recursively walked.
 pub async fn format(
-  args: Vec<String>,
+  args: Vec<PathBuf>,
   check: bool,
-  exclude: Vec<String>,
+  exclude: Vec<PathBuf>,
 ) -> Result<(), AnyError> {
-  if args.len() == 1 && args[0] == "-" {
+  if args.len() == 1 && args.get(0).unwrap().to_str().unwrap() == "-" {
```
Comment (bartlomieju) [inline cli/fmt.rs:39]: `args[0].display()`?

File: cli/lint.rs [hunk 3]
```diff
@@ -40,11 +40,12 @@ fn create_reporter(kind: LintReporterKind) -> Box<dyn LintReporter + Send> {
...
 
 pub async fn lint_files(
-  args: Vec<String>,
-  ignore: Vec<String>,
+  args: Vec<PathBuf>,
+  ignore: Vec<PathBuf>,
   json: bool,
 ) -> Result<(), AnyError> {
-  if args.len() == 1 && args[0] == "-" {
+  if args.len() == 1 && args.get(0).unwrap().to_str().unwrap() == "-" {
```
Comment (bartlomieju) [inline cli/lint.rs:47]: `args[0].display()`

File: cli/fmt.rs [hunk 4]
```diff
@@ -243,11 +243,11 @@ pub fn collect_files(
     ));
   } else {
     for arg in files {
-      let p = PathBuf::from(arg);
-      if p.is_dir() {
-        target_files.extend(files_in_subtree(p.canonicalize()?, is_supported));
+      if arg.is_dir() {
+        target_files
+          .extend(files_in_subtree(arg.canonicalize()?, is_supported));
```
Comment (bartlomieju) [inline cli/fmt.rs:248]: rename `arg` to `file`

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: cli/fmt.rs (hunk 2 above)
Comment (humancalico) [reply cli/fmt.rs:39]: Getting this error on using this
```rust
binary operation `==` cannot be applied to type `std::path::Display<'_>`
```

File: cli/lint.rs [hunk 5]
```diff
@@ -40,11 +40,12 @@ fn create_reporter(kind: LintReporterKind) -> Box<dyn LintReporter + Send> {
...
 pub async fn lint_files(
-  args: Vec<String>,
-  ignore: Vec<String>,
+  args: Vec<PathBuf>,
+  ignore: Vec<PathBuf>,
   json: bool,
 ) -> Result<(), AnyError> {
-  if args.len() == 1 && args[0] == "-" {
+  if args.len() == 1 && args[0].to_string_lossy() == "-" {
+    // TODO don't use unwrap here
```
Comment (bartlomieju) [inline cli/lint.rs:48]: this TODO is outdated

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @humancalico

**Pull request merged**